--- android.toolchain.cmake.bak	2018-02-10 02:01:37.176345700 +0900
+++ android.toolchain.cmake	2018-02-10 02:02:28.059357900 +0900
@@ -717,3 +717,19 @@
 if(ANDROID_TOOLCHAIN STREQUAL clang)
 	set(ANDROID_COMPILER_IS_CLANG TRUE)
 endif()
+
+# CMake 3.7+ compatibility.
+if (CMAKE_VERSION VERSION_GREATER 3.7.0)
+      set(CMAKE_ANDROID_NDK ${ANDROID_NDK})
+      if(ANDROID_TOOLCHAIN STREQUAL gcc)
+             set(CMAKE_ANDROID_NDK_TOOLCHAIN_VERSION 4.9)
+      else()
+              set(CMAKE_ANDROID_NDK_TOOLCHAIN_VERSION clang)
+      endif()
+      set(CMAKE_ANDROID_STL_TYPE ${ANDROID_STL})
+      if(ANDROID_ABI MATCHES "^armeabi(-v7a)?$")
+              set(CMAKE_ANDROID_ARM_NEON ${ANDROID_ARM_NEON})
+              set(CMAKE_ANDROID_ARM_MODE ${ANDROID_ARM_MODE})
+      endif()
+endif()
+
